src/DotLiquid/StandardFilters.cs

```diff
@@ -640,16 +640,27 @@ public static string Date(Context context, object input, string format)
                 return input.ToString();
 
             DateTimeOffset dateTimeOffset;
+            long timestamp = 0L;
             if (input is DateTimeOffset inputOffset)
             {
                 dateTimeOffset = inputOffset;
             }
-            else if ((input is decimal) || (input is double) || (input is float) || (input is int) || (input is uint) || (input is long) || (input is ulong) || (input is short) || (input is ushort))
+            else if ((input is decimal) || (input is double) || (input is float) || (input is int) || (input is uint) || (input is long) || (input is ulong) || (input is short) || (input is ushort)
+                || (input is string stringInput && long.TryParse(stringInput, NumberStyles.Integer, CultureInfo.InvariantCulture, out timestamp)))
             {
-#if CORE
-                dateTimeOffset = DateTimeOffset.FromUnixTimeSeconds(Convert.ToInt64(input)).ToLocalTime();
+                // If input is a string timestamp we already have a long value, if not convert
+                timestamp = input is string ? timestamp : Convert.ToInt64(input);
+
+                // Check if the timestamp is outside the range supported by the DateTimeOffset.FromUnixTimeSeconds() method
+                var isMilliseconds = timestamp < -62_135_596_800 || timestamp > 253_402_300_799;

```

Comment:
So it's impossible to specify a date prior to 1/11/1978 in milliseconds. It's sort of arbitrary IMHO. Would approve without this idea.

Reply:
It's not quite arbitrary, the numbers are taken from the documented range supported by [DateTimeOffset.FromUnixTimeSeconds](https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.fromunixtimeseconds?view=net-5.0#exceptions).

My use case is to parse a UTC epoch milliseconds timestamp and convert to a specified timezone. So, perhaps a new extended filter (such as `MillisToDate`) similar to `Date` with a target date-format parameter, plus a timezone parameter would be a cleaner solution.

Reply:
Agreed on a separate filter. We can't add to the library of you want to specify a timezone as that would require reliance on an external library like NodaTime. I believe there was a library that did just that need to find it.

Reply:
Alternate filter ConvertTime added